### PR TITLE
Fix installing from a readonly copy of synapse

### PR DESCRIPTION
**Pull request by richvdh**
It seems that, as of pip 20.1, pip likes to do an in-tree build of synapse when
you install it, which is pretty nasty when pip is running in a docker image.

We can work around this by copying the bits we need into a tarball and
installing from that. Ultimately I'd like to change the build pipelines to
build a wheel upfront, and use that for the sytest builds, which will be a more
accurate representation of how most people install synapse, but this is a
prerequisite.

See also: https://discuss.python.org/t/installing-from-a-read-only-source/4115/3

### CODE REVIEW

**Comment by richvdh** [comment]
(worth noting that this only becomes a problem if/when we rebuild the sytest-synapse docker images)

File: scripts/synapse_sytest.sh [hunk 1]
```diff
@@ -26,10 +26,10 @@ if [ -n "$MULTI_POSTGRES" ] || [ -n "$POSTGRES" ]; then
...
     su -c 'eatmydata /usr/lib/postgresql/*/bin/pg_ctl -w -D $PGDATA start' postgres
 
-    su -c psql postgres <<< "show config_file"
-    su -c psql postgres <<< "show max_connections"
-    su -c psql postgres <<< "show full_page_writes"
-    su -c psql postgres <<< "show fsync"
+    #su -c psql postgres <<< "show config_file"
+    #su -c psql postgres <<< "show max_connections"
+    #su -c psql postgres <<< "show full_page_writes"
+    #su -c psql postgres <<< "show fsync"
```
Comment (clokep) [inline scripts/synapse_sytest.sh:32]: Should we remove these if they're not useful?

File: scripts/synapse_sytest.sh [hunk 2]
```diff
@@ -110,6 +110,24 @@ elif [ -n "$POSTGRES" ]; then
 
 fi
 
+# default value for SYNAPSE_SOURCE
+: ${SYNAPSE_SOURCE:=/src}
+
+# if we're running against a source directory, turn it into a tarball.  pip
+# will then unpack it to a temporary location, and build it.  (As of pip 20.1,
+# it will otherwise try to build it in-tree, which means writing changes to the
+# source volume outside the container.
```
Comment (clokep) [inline scripts/synapse_sytest.sh:119]:
```suggestion
# source volume outside the container.)
```

File: scripts/synapse_sytest.sh [hunk 3]
```diff
@@ -110,6 +110,24 @@ elif [ -n "$POSTGRES" ]; then
 
 fi
 
+# default value for SYNAPSE_SOURCE
+: ${SYNAPSE_SOURCE:=/src}
```
Comment (clokep) [inline scripts/synapse_sytest.sh:114]: Neat, did not know this about bash. (https://wiki.bash-hackers.org/syntax/pe#assign_a_default_value if anyone else doesn't know what this does)

File: scripts/synapse_sytest.sh (hunk 1 above)
Comment (richvdh) [reply scripts/synapse_sytest.sh]: done in 0165ef9b

**Pull request merged**